.azure-pipelines/template-commonlib.yml

```diff
@@ -1,40 +1,59 @@
+- ${{ each arch in parameters.archs }}:
+  - job: Build_${{ arch }}
+    timeoutInMinutes: 120
+    ${{ if eq(arch,'amd64') }}:
+      pool: sonicbld
+    ${{ else }}:
+      pool: sonicbld-${{ arch }}
+    variables:
+    - template: template-variables.yml
+    steps:
+    - checkout: self
+      clean: true
+      submodules: recursive
+    - script: |
+        set -ex
+        branch=$(Build.SourceBranchName)
+        # replace all: '-' -> '_'
+        branch=${branch//-/_}
+        # replace all: 'a' -> 'A', DIST_MASTER is set in variable.
+        BRANCH=DIST_${branch^^}
+        # variable name is dynamic
+        bldenvs=${!BRANCH}
+        [ "$bldenvs" == "" ] && bldenvs="$(COMMON_LIB_BUILD_ENVS)"
+        for bldenv in $bldenvs
+        do
+          BLDENV=$bldenv make -f Makefile.work configure PLATFORM_ARCH=${{ arch }} PLATFORM=vs ENABLE_DOCKER_BASE_PULL=y
+        done
+        set +x
+        echo "##vso[task.setvariable variable=bldenvs;]$bldenvs"
+      displayName: Make configure
+    - script: |
+        set -ex
+        for bldenv in $(bldenvs)
+        do
+          LIBNL3_VERSION_BASE=$(grep "LIBNL3_VERSION_BASE =" rules/libnl3.mk | awk '{print$3}')
+          LIBNL3_VERSION=$(grep "LIBNL3_VERSION =" rules/libnl3.mk | awk '{print$3}' | sed -e "s/(//" -e "s/)//" -e "s/\\$//" -e "s/LIBNL3_VERSION_BASE/$LIBNL3_VERSION_BASE/")
+          SONIC_BUILD_JOBS=$(nproc) BLDENV=$bldenv make -f Makefile.work target/debs/$bldenv/libnl-3-200_${LIBNL3_VERSION}_${{ arch }}.deb ENABLE_DOCKER_BASE_PULL=y
 
-        LIBYANG_VERSION_BASE=$(grep "LIBYANG_VERSION_BASE =" rules/libyang.mk | awk '{print$3}')
-        LIBYANG_VERSION=$(grep "LIBYANG_VERSION =" rules/libyang.mk | awk '{print$3}' | sed -e "s/\\$//" -e "s/(//" -e "s/)//" -e "s/LIBYANG_VERSION_BASE/$LIBYANG_VERSION_BASE/")
-        SONIC_BUILD_JOBS=$(nproc) BLDENV=$bldenv make -f Makefile.work target/debs/$bldenv/libyang_${LIBYANG_VERSION}_amd64.deb
-      done
-      mv target $(Build.ArtifactStagingDirectory)
-    displayName: Make common lib packages
-  - publish: $(Build.ArtifactStagingDirectory)
-    artifact: common-lib
+          LIBYANG_VERSION_BASE=$(grep "LIBYANG_VERSION_BASE =" rules/libyang.mk | awk '{print$3}')
+          LIBYANG_VERSION=$(grep "LIBYANG_VERSION =" rules/libyang.mk | awk '{print$3}' | sed -e "s/\\$//" -e "s/(//" -e "s/)//" -e "s/LIBYANG_VERSION_BASE/$LIBYANG_VERSION_BASE/")
+          SONIC_BUILD_JOBS=$(nproc) BLDENV=$bldenv make -f Makefile.work target/debs/$bldenv/libyang_${LIBYANG_VERSION}_${{ arch }}.deb
+        done
+        mv target $(Build.ArtifactStagingDirectory)
+      displayName: Make common lib packages
+    - ${{ if eq(arch,'amd64') }}:
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: common-lib
+    - ${{ else }}:
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: common-lib.${{ arch }}
```

Comment:
Why not always copy to common-lib.${{ arch }} ?

Reply:
> Why not always publish to common-lib.${{ arch }} ?

Reply:
sonic-swss-common is using common-lib.
If we change artifact name, it will have impact on sonic-swss-common pipeline.